e3db/client.py

```diff
@@ -893,14 +921,154 @@ def revoke(self, record_type, reader_id):
         if reader_id == self.client_id:
             return
 
-        url = self.__get_url("v1", "storage", "policy", str(self.client_id), str(self.client_id), str(reader_id), record_type)
-        json = {
+        deny_read = {
             'deny': [
                 {
                     'read': {}
                 }
             ]
         }
-        response = requests.put(url=url, json=json, auth=self.e3db_auth)
-        self.__response_check(response)
+
+        self.__put_policy(str(self.client_id), str(self.client_id), str(reader_id), record_type, deny_read)
         self.__delete_access_key(self.client_id, self.client_id, reader_id, record_type)
+
+    def add_authorizer(self, record_type, authorizer_id):
+        """
+        Public Method to share a record type with an authorizer.
+
+        Parameters
+        ----------
+        record_type: str
+            type of the record to be stored
+
+        authorizer_id : str
+            The authorizer's client id (UUID) to share this record type with.
+
+        Returns
+        -------
+        None
+        """
+
+        ak = self.__get_access_key(str(self.client_id), str(self.client_id), str(self.client_id), record_type)
+        if ak is None:
```

Comment:
Writer should put the `ak` for themselves here, as well. Otherwise the writer will generate a new AK when they write a record.

Good catch tho - I think the Java SDK will fail here if no records have been written yet.

Reply:
Addressed in 
e9c712a